suprb/optimizer/solution/ga/base.py

```diff
@@ -78,30 +110,291 @@ def _optimize(self, X: np.ndarray, y: np.ndarray):
                 n=self.population_size,
                 random_state=self.random_state_,
             )
+
             # Note that this expression swallows the last element, if `population_size` is odd
             parent_pairs = map(lambda *x: x, *([iter(parents)] * 2))
 
             # Crossover
-            children = list(
-                flatten(
-                    [
-                        (
-                            self.crossover(A, B, random_state=self.random_state_),
-                            self.crossover(B, A, random_state=self.random_state_),
-                        )
-                        for A, B in parent_pairs
-                    ]
-                )
-            )
-            # If `population_size` is odd, we add the solution not selected for reproduction directly
-            if self.population_size % 2 != 0:
-                children.append(parents[-1])
+            children = self.crossover_children(parent_pairs)
 
             # Mutation
-            mutated_children = [self.mutation(child, random_state=self.random_state_) for child in children]
+            mutated_children = self.mutation_children(X, y, children)
```

Comment:
why X and y?

Reply:
Because `mutate_children` for `SelfAdaptingGeneticAlgorithm2 `needs X and y, but since we are moving that back into the saga directory, I will change it back